### DISCUSSION TITLE
新版直播间删除了“进度同步”功能，不知能否在BE中实现这个功能

### DISCUSSION THREAD

**Discussion by GalaxySnail** [the1812/Bilibili-Evolved#1342] [功能建议 / Ideas]
![QQ截图20201220214206](https://user-images.githubusercontent.com/53637421/102714959-1f146c00-430d-11eb-9408-91a1fede2210.png)

如题。个人实际体验来讲效果还是十分明显的，开了追帧以后延迟可能低至不到1秒，没了这个功能通常就有好几秒延迟。不太清楚用网页脚本实现的难度有多大

**Comment by the1812** [comment]
能拿到已缓冲的视频/音频时间, 不过我也不确定能不能靠这些做出追帧来...

**Comment by GalaxySnail** [comment]
据我了解，追帧大概就是加速播放已缓冲的视频，根据我之前的感觉，具体加速的倍率和延迟的高低有关系，我猜应该是由当前已缓冲的视频时长决定的。也就是说大概需要拿到已缓冲的视频时长、以及加速播放视频。要是有相对简单的办法加速播放的话感觉还好，不过我不清楚直播的播放器和普通的播放器会不会不一样，要是加速播放需要对播放器做很复杂的hack的话那就非常困难了

**Comment by the1812** [comment]
视频都是标准的视频, 就是一些数值不知道b站之前设了多少, 得慢慢去试:
- 已缓冲的时间超过当前时间多少秒时, 开始追帧
- 追帧速率加到多少 (我印象里大概 1.1x~1.2x ?)
- 追到时间差剩多少时停止, 肯定不能直接追完, 不然追完马上弹 loading

**Comment by GalaxySnail** [comment]
具体数值啊，那确实得试。其中追到差多少时停止我觉得可能要受到网速波动范围的影响。
另外，我个人觉得可以弄一套通常情况适用的数值，同时允许用户自定义设置，毕竟不同人、不同时间的网络状况可能不一样
